src/main/views/includes/macros/partyDetails.njk

```diff
@@ -0,0 +1,195 @@
+        <h2 class="heading-medium">{{ t('Your residential address') }}</h2>
+        <p>{{ t("This must be the address where you live.") }}</p>
+         {{ addressInput(form, 'address') }}
+        <h2 class="heading-medium">{{ t('Correspondence address') }}</h2>
+
+        {{ radioGroup(
+          t('Would you like correspondence sent to a different address?'),
+          'hasCorrespondenceAddress',
+          [
+            { label: 'No', value: false },
+            { label: 'Yes, add a correspondence address', value: true }
+          ],
+          form,
+          isInline = false
+        )
+      }}
+
+      <div class="panel panel-border-narrow js-hidden" id="hasCorrespondenceAddress-true" aria-hidden="false">
+        <p>{{ t('All correspondence will be sent here, instead of to your residential address. This is sometimes known as an &lsquo;address for service&rsquo;.') | safe }}</p>
+        {{ addressInput(form, 'correspondenceAddress') }}
+      </div>
+        {{ saveAndContinueButton() }}
+  </form>
+{% endmacro %}
+
+{% macro soleTraderForm(form, csrfToken) %}
+  <form method="post">
+    {{ csrfProtection(csrfToken) }}
+        {{ textInput(t('Full name'), 'name', form, autocomplete='name') }}
+        {{ textInput(t('Business name (optional)'), 'businessName', form) }}
+        <h2 class="heading-medium">{{ t('Your residential address') }}</h2>
+        <p>{{ t("This must be the address where you live.") }}</p>
+        {{ addressInput(form, 'address') }}
+        <h2 class="heading-medium">{{ t('Correspondence address') }}</h2>
+
+        {{ radioGroup(
+          t('Would you like correspondence sent to a different address?'),
+          'hasCorrespondenceAddress',
+          [
+            { label: 'No', value: false },
+            { label: 'Yes, add a correspondence address', value: true }
+          ],
+          form,
+          isInline = false
+        )
+      }}
+
+      <div class="panel panel-border-narrow js-hidden" id="hasCorrespondenceAddress-true" aria-hidden="false">
+        <p>{{ t('All correspondence will be sent here, instead of to your residential address. This is sometimes known as an &lsquo;address for service&rsquo;.') | safe }}</p>
+        {{ addressInput(form, 'correspondenceAddress') }}
+      </div>
+        {{ saveAndContinueButton() }}
+  </form>
+{% endmacro %}
+
+{% macro companyForm(form, csrfToken) %}
+  <form method="post">
+    {{ csrfProtection(csrfToken) }}
+        {{ textInput(t('Company name'), 'name', form, autocomplete='companyName') }}
+        {{ textInput(t('Contact person'), 'contactPerson', form) }}
+        <h2 class="heading-medium">{{ t('Company address') }}</h2>
+        <p>{{ t("Enter the company’s registered office or a company address that has a connection with the claim.") }}</p>
+        {{ addressInput(form, 'address') }}
+        <h2 class="heading-medium">{{ t('Correspondence address') }}</h2>
+
+        {{ radioGroup(
```

Comment:
Corespondence address is great candidate for having macros since it is used in many places

Reply:
Extracted as fragment